src/model/transferlog.js

```diff
@@ -9,25 +9,48 @@ const AlreadyFulfilledError = errors.AlreadyFulfilledError
 const MissingFulfillmentError = errors.MissingFulfillmentError
 const debug = require('debug')('ilp-plugin-virtual:store')
 
+const TRANSFER_PREFIX = 'transfer_'
+
 module.exports = class TransferLog {
 
   constructor (opts) {
     this._store = opts.store
+    this._storeCache = {}
+  }
+
+  _get (transferId) {
+    debug('_get', transferId)
+    return this._storeCache[transferId]
+  }
+
+  _put (id, packaged) {
+    debug('_put', id)
+    this._storeCache[id] = packaged
+  }
 
-    this._cacheItems = {}
-    this._cacheStack = []
-    this._cacheSize = 500
+  drop (id) {
+    delete this._storeCache[id]
+  }
+
+  _write (id) {
+    const packaged = this._getPackaged(id)
+    if (packaged.written) return
+
+    packaged.written = true
+    this._storePackaged(packaged)
+    this._store.put(TRANSFER_PREFIX + id, JSON.stringify(packaged))
+  }
 
-    this.incoming = 'incoming'
-    this.outgoing = 'outgoing'
+  getStoredBalance () {
+    return this._storeBalance
   }
 
-  * get (transferId) {
-    return (yield this._getPackaged(transferId)).transfer
+  get (transferId) {
+    return this._getPackaged(transferId).transfer
   }
 
-  * getFulfillment (transferId) {
-    const packaged = yield this._getPackaged(transferId)
+  getFulfillment (transferId) {
```

Comment:
Shouldn't this still be async? It shouldn't try to keep all the fulfillments in memory and I would expect this to work even if the server restarted after that transfer was fulfilled